rootfs/etc/nginx/lua/test/balancer/sticky_test.lua

```diff
@@ -263,3 +263,209 @@ describe("Sticky", function()
     end)
   end)
 end)
+
+describe("StickyWildcard", function()
+  before_each(function()
+    mock_ngx({ var = { location_path = "/", host = "dev.test.com" } })
+  end)
+
+  after_each(function()
+    reset_ngx()
+  end)
+
+  local test_backend = get_test_backend()
+  local test_backend_endpoint= test_backend.endpoints[1].address .. ":" .. test_backend.endpoints[1].port
+
+  describe("balance()", function()
+    local mocked_cookie_new = cookie.new
+
+    before_each(function()
+      package.loaded["balancer.sticky"] = nil
+      sticky = require("balancer.sticky")
+    end)
+
+    after_each(function()
+      cookie.new = mocked_cookie_new
+    end)
+
+    context("when client doesn't have a cookie set and location is in cookie_locations", function()
+      it("picks an endpoint for the client", function()
+        local sticky_balancer_instance = sticky:new(test_backend)
+        local peer = sticky_balancer_instance:balance()
+        assert.equal(peer, test_backend_endpoint)
+      end)
```

Comment:
How's this test different than https://github.com/kubernetes/ingress-nginx/pull/4478/files#diff-f62b36fb1d17870f3fbc347d5624fc1fR86?